pkg/reconciler/apis/identitycache/api_export_identity_controller.go

```diff
@@ -61,10 +65,27 @@ func NewApiExportIdentityProviderController(
 	queue := workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), ControllerName)
 
 	c := &controller{
-		queue:                        queue,
-		kubeClient:                   kubeClusterClient.Cluster(configshard.SystemShardCluster),
-		configMapLister:              configMapInformer.Lister(),
-		remoteShardApiExportsIndexer: remoteShardApiExportInformer.Informer().GetIndexer(),
+		queue: queue,
+		createConfigMap: func(ctx context.Context, cluster logicalcluster.Name, namespace string, configMap *corev1.ConfigMap) (*corev1.ConfigMap, error) {
+			return kubeClusterClient.Cluster(cluster).CoreV1().ConfigMaps(namespace).Create(ctx, configMap, metav1.CreateOptions{})
+		},
+		getConfigMap: func(cluster logicalcluster.Name, namespace, name string) (*corev1.ConfigMap, error) {
+			return configMapInformer.Lister().ConfigMaps(namespace).Get(clusters.ToClusterAwareKey(cluster, name))
+		},
+		updateConfigMap: func(ctx context.Context, cluster logicalcluster.Name, namespace string, configMap *corev1.ConfigMap) (*corev1.ConfigMap, error) {
```

Comment:
This particular controller is hyper-specific in that it only supports one particular configmap, in one cluster, in one namespace. I opted to keep these as parameters so that this isn't lost in the reconcile impl - as it would be if we closed over that at creation-time? Doing it this way also makes it impossible to mis-initialize the thing.

Reply:
I'd say it's more that I'd never expect someone to code "update(cluster=foo, configMap(cluster=bar))"...but either is fine 😄